gh-136992: Add 'None' as valid SameSite value as per RFC6265bis (#137040)
python/cpython: Doc/library/http.cookies.rst

   in HTTP requests, and is not accessible through JavaScript. This is intended
   to mitigate some forms of cross-site scripting.

   The attribute :attr:`samesite` controls when the browser sends the cookie with
   cross-site requests. This helps to mitigate CSRF attacks. Valid values are
   "Strict" (only sent with same-site requests), "Lax" (sent with same-site
   requests and top-level navigations), and "None" (sent with same-site and
   cross-site requests). When using "None", the "secure" attribute must also

>    be set, as required by modern browsers.

Reviewer: Now I see none. Please just have one blank line.

Author: done,hope this is fine

   in HTTP requests, and is not accessible through JavaScript. This is intended
   to mitigate some forms of cross-site scripting.

   The attribute :attr:`samesite` controls when the browser sends the cookie with
   cross-site requests. This helps to mitigate CSRF attacks. Valid values are
   "Strict" (only sent with same-site requests), "Lax" (sent with same-site
   requests and top-level navigations), and "None" (sent with same-site and
   cross-site requests). When using "None", the "secure" attribute must also
   be set, as required by modern browsers.
   
   The attribute :attr:`partitioned` indicates to user agents that these
   cross-site cookies *should* only be available in the same top-level context
   that the cookie was first set in. For this to be accepted by the user agent,
   you **must** also set ``Secure``.

>    

Reviewer: Well.. now we have three spaces here...

Author: Fixed unintended spaces in blank line .

Reviewer: Once you're done with this line I'll merge (probably tomorrow)

Reviewer: On line 162 there are still 3 whitespaces. You need to have no whitespaces in blank lines. Please allow maintainers to directly commit to your branch in the future, it would prevent such back and forth.

Author: Thank you, for your guidance ! Just to clarify, I had already removed the whitespaces before your approval message came in, so I assume everything’s all good now.
And yes, “Allow edits by maintainers” is already enabled on my PR.

   in HTTP requests, and is not accessible through JavaScript. This is intended
   to mitigate some forms of cross-site scripting.

>    The attribute :attr:`samesite` controls when the browser sends the cookie with
>    cross-site requests. This helps to mitigate CSRF attacks. Valid values are
>    "Strict" (never sent with cross-site requests), "Lax" (sent with top-level
>    navigation), and "None" (always sent). When using "None", the "secure"
>    attribute must also be set, as required by modern browsers.

Reviewer: ```suggestion
   The attribute :attr:`samesite` controls when the browser sends the cookie with
   cross-site requests. This helps to mitigate CSRF attacks. Valid values are
   "Strict" (only sent with same-site requests), "Lax" (sent with same-site
   requests and top-level navigations), and "None" (sent with same-site and
   cross-site requests). When using "None", the "secure" attribute must also
   be set, as required by modern browsers.
```

Let's use the same wording as the RFC.

Author: I’ve made the changes as suggested.
Just wondering , is it okay to slightly rephrase for clarity if the meaning stays the same?

Reviewer: I would prefer using the same wording as in the RFC as their authors likely know better how to word this. But what would have you suggested instead? maybe your wording is even better!

Author: I totally understand the preference for aligning with the RFC wording.
I had initially rephrased it slightly to make it a bit more concise, but I’m happy to switch to the RFC version if that’s preferred. Let me know what works best, and I’ll update accordingly.

Reviewer: Let's keep the RFC version for now.

Author: I’ve updated the SameSite description to follow the RFC wording as discussed.
However, since the update was done via an interactive rebase (to clean up history), my local branch is ahead of the remote. To push the changes, I’ll need to do a git push --force.

Just wanted to double-check if you're okay with a force push in this case ...? I’ll go ahead only if that’s fine with you.

Reviewer: I don't think you needed to change anything. I just wanted you to remove the extra blank lines. I'll commit directly to your branch and merge the stuff myself.

Author: Got it, thank you so much for clarifying.

> 

Reviewer: ```suggestion
```

>    `RFC6265bis <https://datatracker.ietf.org/doc/html/draft-ietf-httpbis-rfc6265bis>`_.

Reviewer 2: Actually, wait, I didn't see @picnixz's comment on the issue. This RFC hasn't been accepted, but per [Mozilla's documentation](https://developer.mozilla.org/en-US/docs/Web/HTTP/Guides/Cookies#controlling_third-party_cookies_with_samesite), we still do need the `secure` attribute. Let's just remove the mention of the RFC.

Author: I've removed the reference to RFC6265bis as suggested. Let me know if there's anything else I should update. Thank you.

   in HTTP requests, and is not accessible through JavaScript. This is intended
   to mitigate some forms of cross-site scripting.

   The attribute :attr:`samesite` controls when the browser sends the cookie with
   cross-site requests. This helps to mitigate CSRF attacks. Valid values are
   "Strict" (only sent with same-site requests), "Lax" (sent with same-site
   requests and top-level navigations), and "None" (sent with same-site and
   cross-site requests). When using "None", the "secure" attribute must also

>    be set, as required by modern browsers.
> 

Reviewer: ```suggestion
   be set, as required by modern browsers.
```

Reviewer: Just commit this suggestion. There is are two blank lines after the end of the paragraph.

Author: I had already applied this fix manually during rebase.

> 

Reviewer: Hum, I don't understand but I still see two blank lines here. L157 and L158

Author: I’ve removed the extra blank line and pushed it as a new commit .
